.github/workflows/main.yml

```diff
@@ -15,26 +15,32 @@ jobs:
         ruby: [jruby, 2.4, 2.5, 2.6, 2.7, 3.0]
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@main
+    - uses: actions/checkout@v2
+
+    # Conditionally configure bundler via environment variables as advised
+    #   * https://github.com/ruby/setup-ruby#bundle-config
+    - name: Set bundler environment variables
+      run: |
```

Comment:
Does this work instead?

```yml
- name: Set bundler environment variables
  env:
    BUNDLE_WITHOUT: "checks:docs"
  if: matrix.ruby != 3.0
```

Reply:
Hi @JacobEvelyn! Thanks this is a good question. Please see for examples [this link](https://github.community/t/possible-to-use-conditional-in-the-env-section-of-a-job/135170/3) documenting that:
* `echo "KEY=VALUE" >> $GITHUB_ENV` runs as a step, and therefore can take on conditional execution using `if:`
* whereas `env:` is a job-level descriptor that has no conditional execution